pkg/proto/nbcontract/v1/customsearchdomain.proto

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message CustomSearchDomain {
+  // Did not find much information on what this really does, but can confirm from the usage in cse_config.sh that these fields are indeed required if set on bootstrapper
+  // They are expanded in the file https://github.com/Azure/AgentBaker/blob/master/parts/linux/cloud-init/artifacts/setup-custom-search-domains.sh
+  // Given that there is a conditional in the script on whether or not these fields are present, these are marked optional, however, in RP datamodel, these arent marked optional as part of Linux Profile
+  optional string name = 1;
+  optional string realm_user = 2;
+  optional string realm_password = 3;
```

Comment:
From the script it seems like all these would be required - if used at all. So maybe it is upper level field that needs to be optional, and these individual fields would be required?

Reply:
oh that makes 100% sense, oversight on my part, let me fix this

Reply:
addressed.